### ISSUE TITLE
Atom 1.32 breaks syntax highlighting

### ISSUE THREAD

**Issue by hermidalc** [atom/language-python#281]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Python syntax highlighting is not working after update to Atom 1.32

### Steps to Reproduce

1. Open any .py file in Atom 1.32

**Expected behavior:** Proper syntax highlighting

**Actual behavior:** Parts of code which should be highlighted are now broken and are white

**Reproduces how often:** 100%

### Versions

Atom 1.32

**Comment by 50Wliu** [comment]
Thanks for contributing! Can you please share some pictures or other describe in further detail the differences in syntax highlighting that you are experiencing?

**Comment by mskornilov** [comment]
It seems like the new Tree-sitter parsing system is causing this weird behaviour.
Turning it off in the settings did the trick for me.
Here's a screenshot with Tree-sitter on:
![screenshot from 2018-10-27 15-34-57](https://user-images.githubusercontent.com/41365290/47604228-15c63800-d9ff-11e8-962c-d092a785ce96.png)
And here's with it off:
![screenshot from 2018-10-27 15-37-05](https://user-images.githubusercontent.com/41365290/47604235-35f5f700-d9ff-11e8-96a0-1928be4faf26.png)

**Comment by Dominik1799** [comment]
@mskornilov  I am sorry to ask this question, however I am pretty new to Atom and this is bothering me for a few days now, how do I turn the   Tree sitter off?

**Comment by hermidalc** [comment]
Thanks @mskornilov for posting pics of the changed syntax highlighting behavior.  Mine looks the same.

**Comment by hermidalc** [comment]
Sorry now that I think about it, which syntax highlighting style (1.32 w/ tree-sitter or 1.31.2 without) is the correct one? Maybe it was broken all this time and now is fixed in 1.32?

**Comment by mskornilov** [comment]
I'd say the old one is correct, at least it properly detects arguments, class methods etc.
tree-sitter:
![screenshot from 2018-10-27 21-22-24](https://user-images.githubusercontent.com/41365290/47607993-175c2400-da30-11e8-9c33-789d698c421a.png)
without:
![screenshot from 2018-10-27 21-22-47](https://user-images.githubusercontent.com/41365290/47607997-204cf580-da30-11e8-9a9f-442650ef2d3b.png)

**Comment by yousseftarek** [comment]
This also extends to JavaScript syntax highlighting. I'm having the same issue here as well.

**Comment by TheElementalOfCreation** [comment]
* `# TODO` is nolonger being highlighted in pink/purple. Now it is just grey.
![image](https://user-images.githubusercontent.com/24580325/47670870-81c5af00-db6b-11e8-97e2-3246fc9c3609.png)
* Don't know if this one was intentional, but typing '.whatever' makes 'whatever' appear as white, and '.' appear as orange.
![image](https://user-images.githubusercontent.com/24580325/47670882-8c804400-db6b-11e8-9ff5-f0eb5e6f21e9.png)
* Triple quotes ( `'''` or `"""` ) do not start highlighting until a close triple quote appears if the first one is not the first character on the line. Instead, they turn everything white after them. They also mess up some of the highlighting before them as well. So if they appear after a space or tab, they do this.
![image](https://user-images.githubusercontent.com/24580325/47670943-b3d71100-db6b-11e8-878a-76222141fdba.png)

* Quotes sometimes just completely crap themselves and stop suddenly after non-letter characters if no end-quote is found
![image](https://user-images.githubusercontent.com/24580325/47670772-43c88b00-db6b-11e8-990a-36b130cb68f7.png)

**Comment by hermidalc** [comment]
If it isn't painfully clear based on what others have said and shown, it's definitely the Tree Sitter Parsers that causing the syntax highlighting problem, and in multiple languages.  In Atom 1.32 if you go to Settings -> Core -> Tree Sitter Parsers and uncheck it highlighting behavior returns to normal.  I'm actually kind of surprised something like this wasn't caught by someone before 1.32 rolled out.

**Comment by Ben3eeE** [comment]
@TheElementalOfCreation 

> TODO is no longer being highlighted in pink/purple. Now it is just grey.
> Brackets in strings are no longer highlighted orange

This is tracked in https://github.com/atom/language-todo/issues/82 & https://github.com/atom/atom/issues/18196

**Comment by maxbrunsfeld** [comment]
The highlighting of keyword arguments has been fixed and the fix will ship in Atom 1.32.1.

**Comment by asiloisad** [comment]
the language highlithing do not deal correctly with greek letters (python 3 fully support unicode characters), so insted of [a-zA-Z] better can be [a-zA-Zα-ωΑ-Ω]

**Comment by maxbrunsfeld** [comment]
I'm going to close this out since the original issue has been fixed. @asiloisad I apologize for breaking the highlighting for identifiers with greek letters. That has been fixed (though the fix has not shipped yet).

@TheElementalOfCreation - The TODO issue is tracked at https://github.com/atom/language-todo/issues/82. The issue where we were highlighting an erroneous `.` as a floating point number 😬 has been fixed on master. I opened a separate issue for the unmatched triple quotes: https://github.com/atom/language-python/issues/285. Thanks for the detailed reports.

Everyone, if there are other issues with python highlighting, please open a new issue that describes the problem specifically. Thank you!

**Comment by hermidalc** [comment]
Hi - am I missing something or is this fix still not available? I have the latest atom with everything updated and when I turn on tree sitter parsers python highlighting looks wrong.

**Comment by hermidalc** [comment]
Sorry want to ask again for some feedback... since I've been using Atom with Tree Sitter Parsers off for so long I am now unsure what would be the correct way Python should be highlighted :-/

Since I wrote my previous comment March 7, Atom or other updates have improved the Tree Sitters Python syntax highlighting to be more correct now.  There still some differences so I'm guessing it was always wrong when Tree Sitters was off (and I just got used to it)?

Tree Sitters ON:
![Screenshot from 2019-04-24 08-32-55](https://user-images.githubusercontent.com/461066/56661273-68f80b00-666f-11e9-886e-9c0df7bbbd24.png)
OFF:
![Screenshot from 2019-04-24 08-33-09](https://user-images.githubusercontent.com/461066/56661307-77debd80-666f-11e9-959c-e71f02755bc8.png)

ON:
![Screenshot from 2019-04-24 08-50-04](https://user-images.githubusercontent.com/461066/56661288-701f1900-666f-11e9-876f-132b502a4146.png)
OFF:
![Screenshot from 2019-04-24 08-49-56](https://user-images.githubusercontent.com/461066/56661326-7f05cb80-666f-11e9-8e7c-e8a4db7a115c.png)

**Comment by 50Wliu** [comment]
@hermidalc function parameters will be highlighted again starting in (I think) Atom 1.37.0 thanks to #297 from @caleb531.

`self` not being highlighted is a separate issue that we haven't tackled yet.

Other variables being highlighted was removed in the tree-sitter grammar as there was no way to highlight all variables consistently (and, indeed, you'll notice the inconsistency in your own examples even with tree-sitter turned off).

**Comment by caleb531** [comment]
@hermidalc Just an update: my improvements to function/class parameter highlighting in #297 and #298 will be actually released in Atom v1.38.0—they are already in beta0, in fact—per the [official release notes](https://atom.io/releases).

If you want them sooner, you can clone my fork of language-python and rebuild apm by running the following:
```
git clone https://github.com/caleb531/language-python.git
cd language-python
apm link
apm rebuild
```

When Atom 1.38.0 is officially released, you can later unlink and rebuild again to use the bundled language-python instead of my fork:
```
cd language-python
apm unlink
apm rebuild
```

Then restart Atom to complete the process.

**Comment by hermidalc** [comment]
@caleb531 highlighting in 1.38 looking good now.  From the above two tree sitters on/off examples I had above the only possible missing hightlight now is in function def if you have e.g. *args or **kwargs the "args" and "kwargs" parts are not orange.

**Comment by 50Wliu** [comment]
That was handled in #303 and should be coming down the release pipeline next :).

**Comment by Arcanemagus** [comment]
> Everyone, if there are other issues with python highlighting, please open a new issue that describes the problem specifically. Thank you!

Please follow these instructions already posted if you run into further issues, thanks!
